projects/observability/src/shared/components/explore-query-editor/order-by/explore-query-order-by-editor.component.ts

```diff
@@ -0,0 +1,239 @@
+  }
+
+  private buildSpecAndEmit(
+    orderByData$: Observable<
+      [AttributeMetadata | undefined, MetricAggregationType | undefined, GraphQlSortDirection | undefined]
+    >
+  ): void {
+    orderByData$
+      .pipe(
+        take(1),
+        switchMap(orderByData => this.buildOrderExpression(...orderByData))
+      )
+      .subscribe(value => this.orderByExpressionChange.emit(value));
+  }
+
+  private getCurrentSelectedAttribute(): Observable<AttributeMetadata | undefined> {
+    return this.metricOptions$.pipe(
+      map(attributeOptions =>
+        attributeOptions.find(option => option.value.name === this.orderByExpression?.keyExpression?.key)
+      ),
+      map(matchedSelection => matchedSelection && matchedSelection.value)
+    );
+  }
+
+  private getCurrentSelectedAggregation(): Observable<MetricAggregationType | undefined> {
+    return this.aggregationOptions$.pipe(
+      map(aggregationOptions =>
+        aggregationOptions.find(option => option.value === this.orderByExpression?.aggregation)
+      ),
+      map(matchedSelection => matchedSelection && matchedSelection.value)
+    );
+  }
+
+  private getCurrentSelectedSortBy(): Observable<GraphQlSortDirection | undefined> {
+    return this.sortByOptions$.pipe(
+      map(sortByOptions => sortByOptions.find(option => option.value === this.orderByExpression?.direction)),
+      map(matchedSelection => matchedSelection && matchedSelection.value)
+    );
+  }
+
+  private getAggregationOptionsForAttribute(selected?: AttributeMetadata): SelectOption<MetricAggregationType>[] {
+    if (selected === undefined) {
+      return [];
+    }
+
+    return selected.allowedAggregations.map(aggType => ({
+      value: aggType,
+      label: getAggregationDisplayName(aggType)
+    }));
+  }
+
+  private getAttributeOptionsForContext(context?: TraceType): Observable<SelectOption<AttributeMetadata>[]> {
+    if (context === undefined) {
+      return of([]);
+    }
+
+    return this.metadataService.getSelectionAttributes(context).pipe(
+      map(attributes =>
+        attributes.map(attribute => ({
+          value: attribute,
+          label: this.metadataService.getAttributeDisplayName(attribute)
+        }))
+      )
+    );
+  }
+
+  private getSortByOptions(): Observable<SelectOption<GraphQlSortDirection | undefined>[]> {
+    return of([
+      {
+        value: undefined,
+        label: 'None'
+      },
+      {
+        value: 'ASC',
+        label: 'Asc'
+      },
+      {
+        value: 'DESC',
+        label: 'Desc'
+      }
+    ]);
+  }
+
+  private buildOrderExpression(
+    attribute?: AttributeMetadata,
+    aggregation?: MetricAggregationType,
+    sortBy?: GraphQlSortDirection
+  ): Observable<GraphQlOrderBy> {
+    if (attribute === undefined || aggregation === undefined) {
+      return EMPTY;
+    }
+
+    const aggregationToUse = attribute.allowedAggregations.includes(aggregation)
```

Comment:
no defaulting please - only default in terms of selection (e.g. I pick some metric, and we default the selector to the first aggregation)

Reply:
done